lib/fluent/plugin/exception_detector.rb

```diff
@@ -190,34 +190,36 @@ class TraceAccumulator
     # message_field may contain the empty string. In this case, the
     # TraceAccumulator 'learns' the field name from the first record by checking
     # for some pre-defined common field names of text logs.
-    def initialize(message_field, languages, &emit_callback)
+    # The optional parameters max_lines and max_bytes limit the maximum amount
```

Comment:
Nit: these aren't optional, they're named. To make them optional, you'd use the `name=value` syntax, i.e.:
```
def initialize(message_field, languages, max_lines=0, max_bytes=0, &emit_callback)
```
which would require you to pass `max_lines` and `max_bytes` as regular (unnamed) arguments, i.e.:
```
Fluent::TraceAccumulator.new(message, @languages, max_lines, max_bytes) do |t, r|
```
Since you intend to specify one or the other, I'd just s/optional/named/. :-)